### PR TITLE
Add SQFC Support

### PR DESCRIPTION
**Pull request by PabstMirror**
Will need to update acre2_tools.zip

### CODE REVIEW

**Comment by PabstMirror** [comment]
- [x]  Look into using https://github.com/arma-actions/sqfc/ (it's exe is out of date right now)

File: addons/main/script_macros.hpp [hunk 1]
```diff
@@ -156,3 +156,9 @@ SQF equivalent of extensions/src/ACRE2Shared/Types.h
 #define BASE_CLASS_CONFIG(configName) call { private _baseClass = getText(configFile >> "CfgWeapons" >> configName >> "acre_baseClass"); if (_baseClass == "") then { _baseClass = getText(configFile >> "CfgVehicles" >> configName >> "acre_baseClass"); }; _baseClass }
 
 #include "script_debug.hpp"
+
+
+// remove after cba 3.15.4 (update script_mod.hpp)
+#ifndef COMPILE_SCRIPT
+#define COMPILE_SCRIPT(var1) compileScript ['PATHTO_SYS(PREFIX,COMPONENT_F,var1)']
+#endif
```
Comment (jonpas) [inline addons/main/script_macros.hpp]: Guess this can be removed now.
```suggestion
```

File: hemtt.toml [hunk 2]
```diff
@@ -65,3 +67,19 @@ steps_linux = [
     "[ -d a3 ] || ln -sr include/a3 a3/"
 ]
 show_output = true
+
+[scripts.compile_sqf]
+steps_windows = [
+    "echo 'compile_sqf'",
```
Comment (jonpas) [inline hemtt.toml]:
```suggestion
    "echo Compile SQF",
```

File: hemtt.toml [hunk 3]
```diff
@@ -65,3 +67,19 @@ steps_linux = [
...
+    "py tools/compile_sqf.py cleanup build"
```
Comment (jonpas) [inline hemtt.toml]:
```suggestion
    "python tools/compile_sqf.py cleanup build"
```

File: hemtt.toml [hunk 4]
```diff
@@ -65,3 +67,19 @@ steps_linux = [
...
+]
+only_release = true
+show_output = true
+
+[scripts.compile_sqf_cleanup]
+steps_windows = [
+    "echo 'compile_sqf_cleanup'",
+    "py tools/compile_sqf.py cleanup"
```
Comment (jonpas) [inline hemtt.toml]:
```suggestion
    "python tools/compile_sqf.py cleanup"
```

File: hemtt.toml [hunk 5]
```diff
@@ -65,3 +67,19 @@ steps_linux = [
...
+steps_windows = [
+    "echo 'compile_sqf'",
+    "py tools/compile_sqf.py cleanup build"
+]
+only_release = true
+show_output = true
+
+[scripts.compile_sqf_cleanup]
+steps_windows = [
+    "echo 'compile_sqf_cleanup'",
```
Comment (jonpas) [inline hemtt.toml]:
```suggestion
    "echo Compile SQF Cleanup",
```

File: sqfc.json [hunk 6]
```diff
@@ -0,0 +1,15 @@
+{
+    "inputDirs": [
+        "P:/idi/acre/addons/"
```
Comment (jonpas) [inline sqfc.json:3]: Release CI has no P-drive.

**Review by jonpas — CHANGES_REQUESTED** [review]

File: sqfc.json (hunk 6 above)
Comment (PabstMirror) [reply sqfc.json:3]: updated release.yml to map one

Comment (jonpas) [reply sqfc.json:3]: Nice

**Review by jonpas — APPROVED** [review]

**Pull request merged**